### PR TITLE
add memgraph_kitty: display memgraphs in the kitty terminal emulator

### PR DESCRIPTION
**Pull request by Armael**

### CODE REVIEW

**Comment by Armael** [comment]
TODO (for tomorrow): write some documentation in `memgraph_kitty.mli`.

**Comment by Armael** [comment]
Done. This is ready for review!

File: memgraph_kitty.opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+version: "1.0"
+synopsis:
+  "Display the representation of memory values in the Kitty terminal emulator"
+description:
+  "Memgraph_kitty inspects ocaml values and displays their graphical representation using the graphics protocol of the kitty terminal emulator"
+maintainer: ["Guillaume Bury <guillaume.bury@gmail.com>"]
```
Comment (Gbury) [inline memgraph_kitty.opam]: You could probably add yourself to the author and/or maintainer list I suppose (not sure how to do that with dune generating these files though).

File: src/kitty/memgraph_kitty.ml [hunk 2]
```diff
@@ -0,0 +1,92 @@
+let font_blob =
```
Comment (Gbury) [inline src/kitty/memgraph_kitty.ml:3]: There should probably be some header added at the top of this file (even if only a one-line comment referring to the LICENSE file like in the other files).

File: dune-project [hunk 3]
```diff
@@ -18,5 +18,19 @@
...
+; Memgraph_kitty package
+(package
+  (name memgraph_kitty)
+  (depends
+     (ocaml (>= 4.05.0))
+     (ppx_blob (and :build (>= 0.7.0)))
+     (nanosvg (>= 0.1))
+     (nanosvg_text (>= 0.1))
+     (kittyimg (>= 0.1))
+     (stb_truetype (>= 0.7))
```
Comment (Gbury) [inline dune-project:35]: you should probably add a dependency on `conf-graphviz` (which already handles the adequate depexts to make sure `dot` is available).

**Review by Gbury — CHANGES_REQUESTED** [review]
Overall this looks good. A few remarks here and there, and once they are fixed, I'll merge this.

I wonder whether it would be possible to have a test of `memgraph_kitty`, but I guess that would be quite complicated to setup.

File: dune-project (hunk 3 above)
Comment (Armael) [reply dune-project:35]: there is! (the line below)

File: src/kitty/memgraph_kitty.ml (hunk 2 above)
Comment (Armael) [reply src/kitty/memgraph_kitty.ml:3]: done I think

File: memgraph_kitty.opam (hunk 1 above)
Comment (Armael) [reply memgraph_kitty.opam]: done

File: dune-project (hunk 3 above)
Comment (Gbury) [reply dune-project:35]: oups, I guess I glossed over that because it didn't have version constraint, my bad, ^^

**Review by Gbury — APPROVED** [review]
Thanks !

**Pull request merged**